File: AWK-Gradle-Planet/app/src/main/java/io/dev/android/gradle/util/global/BaseApplication.java
package io.dev.android.gradle.util.global;

import android.app.Application;
import android.content.Context;

import androidx.multidex.MultiDex;

public class BaseApplication extends Application {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }
}

File: Relic/common/src/main/kotlin/io/common/app/BaseApplication.kt
package io.common.app

import android.app.Application
import android.content.Context

open class BaseApplication : Application() {

    companion object {

        private const val TAG = "BaseApplication"

        lateinit var baseApplication: BaseApplication

        fun getInstance(): BaseApplication {
            return baseApplication
        }

        fun getApplicationContext(): Context {
            return baseApplication.applicationContext
        }
    }

    override fun onCreate() {
        super.onCreate()
        baseApplication = this
    }

}